url7f43a4752f_shalynnho_pokemonfactory_tgz-pJ8-Networking_ServerJ8/src/manager/panel/FactoryProductionManagerPanel.java

```diff
@@ -0,0 +1,325 @@
+package manager.panel;
+
+import java.awt.CardLayout;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Font;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.util.ArrayList;
+
+import javax.swing.BorderFactory;
+import javax.swing.Box;
+import javax.swing.BoxLayout;
+import javax.swing.JPanel;
+import javax.swing.JScrollPane;
+import javax.swing.JSpinner;
+import javax.swing.SpinnerNumberModel;
+
+import manager.FactoryProductionManager;
+import manager.panel.KitsListPanel.KitSelectHandler;
+import manager.util.ClickablePanel;
+import manager.util.ClickablePanelClickHandler;
+import manager.util.CustomButton;
+import manager.util.ListPanel;
+import manager.util.OverlayInternalFrame;
+import manager.util.WhiteLabel;
+import Utils.Constants;
+import factory.KitConfig;
+import factory.Order;
+
+/**
+ * 
+ * @author Shalynn Ho, Harry Trieu, Matt Zecchini, Peter Zhang
+ */
+
+public class FactoryProductionManagerPanel extends OverlayInternalFrame {
+	// Width of the JPanel
+	private static final int PANEL_WIDTH = 300;
+
+	private static final String FULL_PANEL = "full";
+	private static final String SIMPLE_PANEL = "simple";
+
+	// A reference to the FactoryProductionManager client
+	private final FactoryProductionManager fpmClient;
+
+	// Stores the selected kitConfig for a new order
+	private KitConfig selectedKit;
+	// Stores the selected Order
+	private Order selectedOrder;
+
+	/** JComponents **/
+	// Displays current schedule of orders
+	private final SpinnerNumberModel spinnerModel;
+	private final JSpinner quantitySpinner;
+	private final CustomButton orderButton;
+	private final KitsListPanel kitsPanel;
+	private final JScrollPane kitsScrollPane;
+	private final OrdersListPanel ordersPanel;
+	private final JScrollPane ordersScrollPane;
+	private final JPanel quantityOrderPanel;
+
+	private final WhiteLabel currentKitCount;
+	private final WhiteLabel currentOrderCount;
+
+	private final WhiteLabel currentKitText;
+	private final WhiteLabel currentOrderText;
+
+	private final JPanel fullPanel = new JPanel();
+	private final JPanel simplePanel = new JPanel();
+	private final JPanel wrapperPanel = new JPanel();
+	private final CardLayout cl = new CardLayout();
+
+	private final PanelMouseListener panelListener = new PanelMouseListener();
+
+	private final boolean visible = true;
+	private final int height;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param f
+	 *            a reference to the FactoryProductionManager client.
+	 * @param height
+	 *            of the JFrame
+	 */
+	public FactoryProductionManagerPanel(FactoryProductionManager f, int height) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 54 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 98% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 92-218 into a separate method.